### PR TITLE
ICU-22324 Mavenization, publish the root pom to Maven

### PR DESCRIPTION
**Pull request by mihnita**
The fix seems more complicated than it is.

But there are basically 2 small changes that account for all the touched files:
1. Publishing was disabled in root, was inherited by all modules, and turned on in the 3 artifacts we care about (`icu4j`, `charset`, and `localespi`. But because now we want to publish the root artifact we need to disable it in all artifacts, except the 3. This accounts for all `the maven-deploy-plugin` ... `<skip>` changes
2. I removed `main/pom.xml` and skipped one level, so that we don't have to also publish the `main` artifact, just the root. This accounts for all the `<relativePath>../../pom.xml</relativePath>` changes in `<parent>`, and the `main/foo` in the root module.

What's left are the small but more interesting changes to review.

##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22324
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [x] Tests included, if applicable
- [x] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Comment by mihnita** [comment]
Tested by publishing in my own fork and tested that cldr/tools works (after pointing it to my fork)

File: icu4j/pom.xml [hunk 1]
```diff
@@ -36,8 +53,8 @@
     <maven.compiler.source>1.8</maven.compiler.source>
     <maven.compiler.target>1.8</maven.compiler.target>
 
-    <maven_central_repo_url>https://oss.sonatype.org</maven_central_repo_url>
-    <maven-central-releases-repo-url>${maven-central-repo-url}/service/local/staging/deploy/maven2</maven-central-releases-repo-url>
+    <!-- <maven-central-repo-url>https://oss.sonatype.org</maven-central-repo-url> -->
+    <maven-central-repo-url>http://mnita.mtv.corp.google.com:8081/nexus</maven-central-repo-url>
```
Comment (markusicu) [inline icu4j/pom.xml:57]: please fix...

**Review by echeran — DISMISSED** [review]
LGTM, thanks!

File: icu4j/pom.xml (hunk 1 above)
Comment (mihnita) [reply icu4j/pom.xml:57]: Ouch!
Thanks, will do.

**Review by echeran — APPROVED** [review]

File: .github/workflows/maven.yaml [hunk 2]
```diff
@@ -7,7 +7,7 @@ on:
     # https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow
 
 env:
-  SHARED_MVN_ARGS: '--show-version --no-transfer-progress'
+  SHARED_MVN_ARGS: '--show-version'
```
Comment (srl295) [inline .github/workflows/maven.yaml:10]: thanks

**Review by srl295 — APPROVED** [review]

**Comment by pedberg-icu** [comment]
I will probably merge this as soon as I see that it is done (if somebody else does not get to it first...).

**Comment by mihnita** [comment]
> I will probably merge this as soon as I see that it is done (if somebody else does not get to it first...).

Thank you!
I had an eye on it, but you were faster.
M

**Pull request merged**